### PR TITLE
Table prefix column names

### PR DESCRIPTION
**Pull request by hllizi**
Fix for #1035: so far, only the filter functions for joined tables prefixed the column names with the table names. In the present case, this resulted in the column name "id" becoming ambiguous, being a columns in both joined tables.
All the filter functions now prefix the column name with the table name to rule out such ambiguities.

### CODE REVIEW

File: IHP/Pagination/ControllerFunctions.hs [hunk 1]
```diff
@@ -134,4 +134,4 @@ defaultPaginationOptions =
         {
             maxItems = 50
         ,   windowSize = 5
-        }
\ No newline at end of file
+        }
```
[the remark this answers is not in the document]
Comment (hllizi) [reply IHP/Pagination/ControllerFunctions.hs:137]: I can't say much about that at the moment, sorry. I'd need to have a look.

**Comment by hllizi** [comment]
I do not know yet what's wrong with the last failing test case. It doesn't fail in my local environment.

**Comment by hllizi** [comment]
Probably fixed, I just wasn't up to date.

**Comment by hllizi** [comment]
Ah, the other way round, of course. It's failing here, too, after merging. Seems like the tests still need to be adapted to #1037.

**Comment by mpscholten** [comment]
Thanks :) Tests are failing because of my recent changes to master, so nothing on your side :)

Is it possible to only use the prefixes when a joined query builder is used? IMO it makes the query a bit more verbose and the query builder should aim to always generate a query that a human would also type into `psql`

**Comment by hllizi** [comment]
It's a bit funny that I had just filed relative column names under "SQL is a train wreck anyway, don't bother", after asking myself the same thing. And it might really get tricky to do that, because, for one thing, you'd have to differentiate between join query builders and ordinary query builders in the filter functions, which so far only need to be constrained with `HasQueryBuilder`. And I guess you'd need to go backwards, not just forwards: only a join turns an ordinary query builder into a join query builder, but once that happens, all column names should be absolute, and that might involve some serious type-level magic.

There are alternatives,I guess, but they would involve to dealing with Text-representation in the filter functions at all and use abstractions which could later be turned into absolute or relative column names, depending on what's needed. It might even be possible to get really fine-grained there and only ever use absolute column names where really necessary, but is the goal really worth the cost?

**Comment by mpscholten** [comment]
Good point, not worth it 👍

**Comment by mpscholten** [comment]
Thanks :) Good job!

**Comment by CSchank** [comment]
Perhaps if the DSL writing the queries had enough information of the SQL query perhaps it could be smart about when it decides to table-prefix the columns. E.g. if the column name is unique within a query it wouldn't do table-prefix it.

**Pull request merged**